src/Helpers/ApartmentHelpers.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace BuildEmpire\Apartment\Helpers;
+
+class ApartmentHelpers
+{
+    /**
+     * Does the schema only contain letters and/or numbers.
+     *
+     * @param $schemaName
+     * @return bool
+     */
+    public static function isSchemaNameValid($schemaName)
+    {
+        return (preg_match('/^[a-z][a-z0-9_]*$/', $schemaName) && !ctype_upper($schemaName));
+    }
+
+    /**
+     * This will remove any bad characters from the $schemaName variable.
+     *
+     * Note: Removes any non valid strings with #. You should be checking with isSchemaNameValid before reaching this
+     * section. This is just an extra layer of protection.
+     *
+     * @param $schemaName
+     * @return mixed
+     */
+    public static function getSchemaSafeString($schemaName) {
+        return preg_replace('/^[^0-9a-z_]+$/', '#',$schemaName);
```

Comment:
Some of the statements like CREATE SCHEMA and prefixing tables will not go through the normal PDO way.

It's a last minute attempt to secure the sql string before it goes out. All the methods should throw an exception if it's not a valid schema, but as an extract precaution this method is applied just before DB call.

At first I removed the invalid character, but what if this happened:

Good guy:
Company BuildEmpire
Schema: buildempire

Bad Guy:
JonDelHall
Schema: buildempire^

If the end-user was not responsible for removing/checking the schemas he/she is creating then: 

In this case the getSchemaSafeString would return:
buildempire#

In the old case it would return:
buildempire

which would be harmful.